tests/table/test_name_mapping.py

```diff
@@ -238,3 +245,67 @@ def test_mapping_lookup_by_name(table_name_mapping_nested: NameMapping) -> None:
 
     with pytest.raises(ValueError, match="Could not find field with name: boom"):
         table_name_mapping_nested.find("boom")
+
+
+def test_invalid_mapped_field() -> None:
+    with pytest.raises(ValueError):
+        MappedField(field_id=1, names=[])
+
+
+def test_update_mapping_no_updates_or_adds(table_name_mapping_nested: NameMapping) -> None:
+    assert update_mapping(table_name_mapping_nested, {}, {}) == table_name_mapping_nested
+
+
+def test_update_mapping(table_name_mapping_nested: NameMapping) -> None:
+    updates = {1: NestedField(1, "foo_update", StringType(), True)}
+    adds = {
+        -1: [NestedField(18, "add_18", StringType(), True)],
+        15: [NestedField(19, "name", StringType(), True), NestedField(20, "add_20", StringType(), True)],
+    }
```

Comment:
Should we also have checks in place when you add an existing ID?

Reply:
Or when you reassign an existing name?

Reply:
> Or when you reassign an existing name?

I think the "name" case covers the reassignment of the existing name.

> Should we also have checks in place when you add an existing ID?

This is an interesting one: Do you think we should add this case? [UpdateSchema](https://github.com/apache/iceberg-python/blob/main/pyiceberg/table/__init__.py#L1608) won't add existing IDs as it uses the `next(self._last_column_id)` to assign the next field_id when a new column is added. The NameMapping class on its own does not have a check to ensure that the field IDs across the MappedFields is unique, and I wonder if we should bother introducing that contraint at the class level.

Reply:
Ah, I see. That makes sense to me. Thanks!